modules/geo-layers/src/mesh-layer/mesh-layer.js

```diff
@@ -0,0 +1,147 @@
+import {log} from '@deck.gl/core';
+import {SimpleMeshLayer} from '@deck.gl/mesh-layers';
+
+import vs from './mesh-layer-vertex.glsl';
+import fs from './mesh-layer-fragment.glsl';
+
+function validateGeometryAttributes(attributes, useMeshColors) {
+  const hasColorAttribute = attributes.COLOR_0 || attributes.colors;
+  const useColorAttribute = hasColorAttribute && useMeshColors;
+  if (!useColorAttribute) {
+    attributes.colors = {constant: true, value: new Float32Array([1, 1, 1])};
+  }
+  log.assert(
+    attributes.positions || attributes.POSITION,
+    'no "postions" or "POSITION" attribute in mesh'
+  );
+}
+
+/*
+ * Convert mesh data into geometry
+ * @returns {Geometry} geometry
+ */
+function getGeometry(data, useMeshColors) {
+  if (data.attributes) {
+    validateGeometryAttributes(data.attributes, useMeshColors);
+    if (data instanceof Geometry) {
+      return data;
+    }
+    return new Geometry(data);
+  } else if (data.positions || data.POSITION) {
+    validateGeometryAttributes(data, useMeshColors);
+    return new Geometry({
+      attributes: data
+    });
+  }
+  throw Error('Invalid mesh');
+}
+
+const defaultProps = {
+  // flat or pbr
+  _lighting: {type: 'string', value: 'flat'},
+  // PBR material object. _lighting must be pbr for this to work
+  pbrMaterial: {type: 'object', value: null}
+};
+
+export default class _MeshLayer extends SimpleMeshLayer {
+  getShaders() {
+    const {_lighting} = this.props;
+    const shaders = super.getShaders();
+    const modules = shaders.modules;
+
+    if (_lighting === 'pbr') {
+      modules.push(pbr);
+    }
+
+    return {...shaders, vs, fs};
+  }
+
+  updateState({props, oldProps, changeFlags}) {
+    super.updateState({props, oldProps, changeFlags});
+    if (props._lighting !== oldProps._lighting && props._lighting === 'pbr') {
+      this.setPbrMaterial(props.pbrMaterial);
+    }
+  }
+
+  draw({uniforms}) {
+    this.state.model.setUniforms({
+      // Needed for PBR (TODO: find better way to get it)
+      u_Camera: this.state.model.getUniforms().project_uCameraPosition
+    });
+    super.draw({uniforms});
+  }
+
+  getModel(mesh) {
+    let materialParser = null;
+    if (this.props._lighting === 'pbr') {
+      const pbrMaterial = this.props.pbrMaterial;
+      const unlit = Boolean(
+        pbrMaterial.pbrMetallicRoughness && pbrMaterial.pbrMetallicRoughness.baseColorTexture
+      );
+      materialParser = new GLTFMaterialParser(this.context.gl, {
+        attributes: {NORMAL: mesh.attributes.normals, TEXCOORD_0: mesh.attributes.texCoords},
+        material: {unlit, ...pbrMaterial},
+        pbrDebug: false,
+        imageBasedLightingEnvironment: null,
+        lights: true,
+        useTangents: false
+      });
+    }
+
+    const shaders = this.getShaders();
+
+    const model = new Model(this.context.gl, {
+      ...this.getShaders(),
+      id: this.props.id,
+      geometry: getGeometry(mesh, this.props._useMeshColors),
```

Comment:
You don't need this experiment prop in a dedicated layer

Reply:
In other words
- at minimum, we can drop the underscore from `_useMeshColors`, no need to mark the prop as experimental as the `_MeshLayer` itself is experimental.
- Unless we think this prop is very useful, we can also drop the property completely and always use the vertex colors (there is no backwards compatibility issue).
- I assume someone who doesn't want vertex colors could avoid supplying them. So unless we think that users want to dynamically turn vertex colors on and off, we can "drop the prop".

Reply:
Done